### PR TITLE
chore: setup website for docs

### LINKED ISSUE #33: Setup website for docs
#### Perceived Problem

- README will not scale for the kind of detailed and verbose documentation that Nexus Prisma needs
- Many large code blocks needed to span the stack that Nexus Prisma sits across (db<>seed<>api<>client<>query<>json result)

#### Ideas / Proposed Solution(s)

- https://github.com/shuding/nextra
- https://github.com/remorses/dokz
- integrate with GH Pages
- Setup custom domain at https://nexus.prisma.io

### PR DESCRIPTION
**Pull request by iddan**
closes #33

### CODE REVIEW

File: docs/pages/index.md [hunk 1]
```diff
@@ -0,0 +1,923 @@
...
+}
+```
+
+```ts
+import { User } from 'nexus-prisma'
+import { objectType } from 'nexus'
+
+objectType({
+  name: User.$name
+  description: User.$description
```
Comment (koniuszy) [inline docs/pages/index.md:170]:
```suggestion
  name: User.$name,
  description: User.$description,
```

**Comment by jasonkuhrt** [comment]
@iddan I'm ready to look at this this week. Can you share a TL;DR about the change and how this works?

**Comment by iddan** [comment]
The change adds the docs directory. Within it there is a Nextra website initialised with the README text as a markdown page and configured with support for GFM markdown support

**Comment by jasonkuhrt** [comment]
@iddan did you figure out hosting yet? gh-pages?

**Comment by iddan** [comment]
Since it’s Next app without SSR we can export the website to static files and host them in GitHub pages

**Comment by iddan** [comment]
This guide defines how to do it correctly: https://lukaszwozniak.dev/blog/deploy-nextjs-app-to-github-pages-using-github-actions/

**Comment by jasonkuhrt** [comment]
@iddan can you please setup deployment automation. Here's what I want:

1. A GH Action
2. Looks at branch, if trunk, continue, if not, stop
3. Looks for changes to /docs (uses git)
4. If no changes stop
5. If yes changes, then build the website and make a push to the gh-pages branch in order to trigger the deployment


Regarding (2) the issue I see with that is that it doesn't not support preview deployments.

With Vercel we get preview deployments.

If we cannot have preview deployments with gh-pages then I suggest Vercel... I think.

**Comment by jasonkuhrt** [comment]
Sometimes CI will fail us, I would also like an npm script in the repo to manually force-deploy the website too.

**Comment by iddan** [comment]
I don’t think you can get preview deploys with GitHub Pages, only services like Vercel and Netlify offer this feature

**Comment by jasonkuhrt** [comment]
@iddan let's use Vercel then. I can use the Prisma Vercel account for this.

**Comment by jasonkuhrt** [comment]
Done, but in order to proceed I need to merge some kind of docs thing to complete configuration. Right now because this isn't on trunk Verecl will not let me specify `/docs` as the root.

I will try to setup a minimal nextra thing on trunk now.

**Comment by jasonkuhrt** [comment]
@iddan please rebase 

You should start getting preview deployments now...

Production: https://nexus-prisma.vercel.app/

**Comment by iddan** [comment]
Ready to merge

**Review by jasonkuhrt — CHANGES_REQUESTED** [review]
Some things:

- The contents are out of date with the current README on trunk
- The trunk README should become minimal and link to the website

Looking at this:

CleanShot 2021-08-31 at 11 14 56@2x


If any of the following is a quickwin please do:

- Can we have more level of nesting in the left sidebar?
- Can we remove the "Index" parent section, makes no sense
- I guess if we combine the two points above it means each current subsection becomes a top level section and then some new subsections would bubble up
- styling isn't very nice but not sure there is an easy theme we can plug there 
- don't like the code highlighting theme

CleanShot 2021-08-31 at 11 17 28@2x


- Copywrite should be Prisma Data Inc.
- Would like to remove the "edit this page" link

**Comment by jasonkuhrt** [comment]
I setup a new domain for us https://nexus.prisma.io/

**Comment by iddan** [comment]
The text I’ve put there was for demonstration purposes. I was expecting once this PR is merged that the content of the website will be changed as you’d like. I’ll make the changes.

**Comment by iddan** [comment]
- [x] Can we have more level of nesting in the left sidebar?
 - [x] Can we remove the "Index" parent section, makes no sense
 - [x] I guess if we combine the two points above it means each current subsection becomes a top level section and then some new subsections would bubble up
 - [ ] styling isn't very nice but not sure there is an easy theme we can plug there
 - [ ] don't like the code highlighting theme
 - [x] Copywrite should be Prisma Data Inc.
 - [x] Would like to remove the "edit this page" link

Can the PR be merged and improvements to the left points be done in additional PRs?

**Review by jasonkuhrt — APPROVED** [review]
Thanks @iddan!

**Pull request merged**